### PR TITLE
Add check for out of range coordinates when indexing.

### PR DESCRIPTION
**Pull request by isaacbrodsky**
Increase the library's resilience to out of range and defective inputs.

### CODE REVIEW

**Comment by isaacbrodsky** [comment]
I haven't yet debugged why the tests work locally but not in CI, I would assume the problem is we're running into undefined behavior. I might add a test that calls `_faceIjkToH3` directly. A test could also be added to `geoToH3` for out of range coordinates.

**Comment by dfellis** [comment]
It's failing in the test H3Index test. I don't have my laptop this weekend, so I can't check any further why.

File: src/h3lib/lib/h3Index.c [hunk 1]
```diff
@@ -557,6 +557,11 @@ H3Index _faceIjkToH3(const FaceIJK* fijk, int res) {
 
     // check for res 0/base cell
     if (res == 0) {
+        if (fijk->coord.i > 2 || fijk->coord.j > 2 || fijk->coord.k > 2) {
```
Comment (nrabinowitz) [inline src/h3lib/lib/h3Index.c]: Can we have a well-named constant for `2` here?

Comment (joegilley) [reply src/h3lib/lib/h3Index.c]: +1 well-named constant and maybe a comment on why > 2 for any of the IJK is bad too :D ?

File: src/h3lib/lib/h3Index.c [hunk 2]
```diff
@@ -557,6 +557,11 @@ H3Index _faceIjkToH3(const FaceIJK* fijk, int res) {
...
+            // out of range input
+            return 0;
```
Comment (joegilley) [inline src/h3lib/lib/h3Index.c]: is this a new sentinel value? if it is, can we add it to a documentation? if it isn't new, never mind :)

Comment (isaacbrodsky) [reply src/h3lib/lib/h3Index.c]: This is the same sentinel used in geoToH3, which is why I used it here.

**Comment by isaacbrodsky** [comment]
I tracked the failure to a difference in `_geoDistRads` when compiling under GCC 4.8, causing a difference in which face is selected for the out of bounds coordinates. I'll determine why that is different next, or change the test to not depend on it.

**Pull request merged**